### DISCUSSION TITLE
Add a heat source with a specified temperature

### DISCUSSION THREAD

**Discussion by 0Clark** [AMReX-Combustion/PeleLMeX#384] [Q&A]
Hello,
I am a newcomer to Pele and I want to set an ignition source with specific temperature in **PeleLMex**, which has been done in **PeleC,** https://github.com/AMReX-Combustion/PeleC/discussions/384?sort=top, and https://github.com/AMReX-Combustion/PeleC/discussions/789.  How to set it in **PeleLMex**? Is there a tutorial for this? I hope to receive advice.
Thanks a lot.

**Comment by drummerdoc** [comment]
There is no tutorial.  Note that LMeX's energy variables are different (enthalpy vs total energy) and also that the expression for divu will need to incorporate the energy source term in a consistent way as well. So, your additions will require some care, and will depend on the form of the equation-of-state to get the divu expression correct.

**Comment by baperry2** [comment]
To guide your efforts, you can follow what is done for soot and sprays (see, e.g. implementation and call locations of `computeSootSource`, which compute source terms that get added to `m_extSource`). See also #246.

**Comment by ThomasHowarth** [comment]
Not sure if you've managed to implement this @0Clark , but I have an unchecked in function to do exactly this - I'll work on tidying it up for a pull request.

**Reply by drummerdoc** [reply]
@ThomasHowarth I'm curious to know if this has turned out to be reasonably robust.  I've had interactions recently with several collaborators in different context where I've hacked up an ignition process that required a little bit of futzing (of the sort that is easy for an old hack but hard to describe to someone new).  It would be awesome if your solution ends up being useful for new folks to get a flame going almost independent of the chemistry mech they are using.

**Reply by ThomasHowarth** [reply]
I think it's reasonably robust. There's no dependence on chemistry, barring some EOS calls, just a source term for the enthalpy equation which already gets carried through to the [divu calculation](https://github.com/AMReX-Combustion/PeleLMeX/blob/fcaf8157a3d8351a63e8d78f0a01a413b890f43c/Source/PeleLMeX_K.H#L204C1-L204C50). It's perhaps a bit clipped in terms of how it's added in spatially and temporally, but that could be massaged.

**Comment by drummerdoc** [comment]
In the past, there hasn't been explicit dependencies on chemistry, but the sensitivity of settings definitely varied with the detail in the mechanism.  drm19 ignites like a champ, but some of the larger, stiffer models can be irritating.  It could be interesting to see if it helps to ignite a SAF flame I'm playing with.
